source/bson-corpus/tests/document.json

```diff
@@ -17,6 +17,26 @@
             "description": "Single-character key subdoc",
             "bson": "160000000378000E0000000261000200000062000000",
             "extjson": "{\"x\" : {\"a\" : \"b\"}}"
+        },
+        {
+            "description": "DBRef",
+            "bson": "37000000036462726566002b0000000224726566000b000000636f6c6c656374696f6e00072469640058921b3e6e32ab156a22b59e0000",
+            "extjson": "{\"dbref\": {\"$ref\": \"collection\", \"$id\": {\"$oid\": \"58921b3e6e32ab156a22b59e\"}}}"
+        },
+        {
+            "description": "DBRef with database",
+            "bson": "4300000003646272656600370000000224726566000b000000636f6c6c656374696f6e00072469640058921b3e6e32ab156a22b59e0224646200030000006462000000",
+            "extjson": "{\"dbref\": {\"$ref\": \"collection\", \"$id\": {\"$oid\": \"58921b3e6e32ab156a22b59e\"}, \"$db\": \"db\"}}"
+        },
+        {
+            "description": "DBRef with additional fields",
+            "bson": "4400000003646272656600380000000224726566000b000000636f6c6c656374696f6e00072469640058921b3e6e32ab156a22b59e02666f6f0004000000626172000000",
+            "extjson": "{\"dbref\": {\"$ref\": \"collection\", \"$id\": {\"$oid\": \"58921b3e6e32ab156a22b59e\"}, \"foo\": \"bar\"}}"
```

Comment:
Moved DBRef test cases into their own dbref.json file.

Added another test case for DBRef with database + other fields. The id in this case is an int, not an ObjectId. Good call on the additional test.